File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> mp;

        vector<vector<int>> vec;

        vector<int> ans;

        for(int n: nums){
            mp[n]++;
        }

        for(auto it: mp){
            vec.push_back({it.second,it.first});
        }

        sort(vec.begin(), vec.end(),[](vector<int>&a, vector<int>&b){
            return a[0]>b[0];
        });

        for(int i=0; i<k; i++){
            ans.push_back(vec[i][1]);
        }
        return ans;
        
    }
};

File: Data Structures & Algorithms/continuous-subarray-sum/submission-3.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        
        prefix = 0
        mp = {0:-1}

        for i,n in enumerate(nums):

            prefix += n

            if prefix%k in mp and i-mp[prefix%k]>=2:
                return True

            elif prefix%k not in mp:   
                mp[prefix%k] = i

        return False